### PR TITLE
DocumentationAPI: Add `MalformedComment`

### LINKED ISSUE #4548: DocumentationAPI Bug with `C` test cases.
We now use `markers.split()` to find docstrings. Amend the changes to `assemble()`
https://github.com/coala/coala/pull/4530/files#r128902524
Opened by @damngamerz at [gitter//damngamerz](https://gitter.im//damngamerz?at=597591db614889d475e41f4e)

### PR DESCRIPTION
**Pull request by damngamerz**
`MalformedComment` is the new error handling mechanism
in DocumentationAPI. Which will help in yielding a
subsequent `RESULT` with a beautiful message in the bear.

Closes #4548

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by Nosferatul** [comment]
looks good to me :+1:

**Comment by damngamerz** [comment]
For reference https://github.com/coala/coala-bears/pull/2001

File: coalib/bearlib/languages/documentation/DocumentationComment.py [hunk 1]
```diff
@@ -284,3 +284,23 @@ def assemble(self):
...
+
+    def __init__(self, message, line):
+        """
+        Instantiate a MalformedComment, which contains the information about
+        the errors generated by DocumentationAPI.
+
+        :param message:
+            Contains the message about the error.
+        :param line:
+            Contains the related line no.
```
Comment (SanketDG) [inline coalib/bearlib/languages/documentation/DocumentationComment.py]: please be more explicit

Comment (SanketDG) [inline coalib/bearlib/languages/documentation/DocumentationComment.py]: what line no and all

File: coalib/bearlib/languages/documentation/DocumentationComment.py [hunk 2]
```diff
@@ -284,3 +284,23 @@ def assemble(self):
...
+class MalformedComment:
+    """
+    The MalformedComment holds information about the errors generated by the
+    DocumentationAPI.
+    """
+
+    def __init__(self, message, line):
+        """
+        Instantiate a MalformedComment, which contains the information about
+        the errors generated by DocumentationAPI.
```
Comment (SanketDG) [inline coalib/bearlib/languages/documentation/DocumentationComment.py]: DocumentationAPI is a very internal term, explain this

File: coalib/bearlib/languages/documentation/DocumentationExtraction.py [hunk 3]
```diff
@@ -197,7 +197,16 @@ def _extract_doc_comment_from_line(content, line, column, regex,
                 doc = DocumentationComment(documentation, docstyle_definition,
                                            indent, marker, position)
 
-                return end_line, end_column, doc
+                break
+
+        if doc_comment:
+            return end_line, end_column, doc
+        else:
+            doc = MalformedComment(dedent("""\
```
Comment (SanketDG) [inline coalib/bearlib/languages/documentation/DocumentationExtraction.py]: `doc`? that's a bad variable name :P

File: tests/bearlib/languages/documentation/DocBaseClassTest.py [hunk 4]
```diff
@@ -301,3 +302,40 @@ def test_DocBaseClass_process_documentation_not_implemented(self):
...
+    def test_MalformedComment1_C(self):
+        data = ['/**\n',
+                '* A doc-comment aborted in the middle of writing\n',
+                '* This won\'t get parsed (hopefully...)\n']
+
+        for doc_comment in DocBaseClass.extract(data, 'C', 'doxygen'):
+            self.assertEqual(
+                [doc_comment.message, doc_comment.line],
+                [dedent("""\
+                    Please check the docstring for faulty markers. A starting
```
Comment (SanketDG) [inline tests/bearlib/languages/documentation/DocBaseClassTest.py]: store the string already in a variable :P

File: coalib/bearlib/languages/documentation/DocumentationComment.py [hunk 5]
```diff
@@ -284,3 +284,29 @@ def assemble(self):
...
+    def __init__(self, message, line):
+        """
+        Instantiate a MalformedComment, which contains the information about
+        the error: a message explaining the behaviour and a line no where the
+        error has occured.
+
+        :param message:
+            Contains the message about the error.
+        :param line:
+            Contains the current line of the docstring where the error has
```
Comment (SanketDG) [inline coalib/bearlib/languages/documentation/DocumentationComment.py]: https://github.com/coala/coala/issues/4644

**Comment by SanketDG** [comment]
ack c5da58e bd80892

**Comment by SanketDG** [comment]
rebase and get it merged

**Comment by gureks** [comment]
@rultor merge please

**Comment by gureks** [comment]
ack 35750ae 0f81583

**Comment by gureks** [comment]
@rultor please merge

**Pull request merged**